torchaudio/csrc/decoder/src/decoder/LexiconFreeDecoder.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in
+ * https://github.com/flashlight/flashlight/blob/d385b2150872fd7bf106601475d8719a703fe9ee/LICENSE
+ */
+
+#include <stdlib.h>
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <numeric>
+
+#include "torchaudio/csrc/decoder/src/decoder/LexiconFreeDecoder.h"
+
+namespace torchaudio {
+namespace lib {
+namespace text {
+
+void LexiconFreeDecoder::decodeBegin() {
+  hyp_.clear();
+  hyp_.emplace(0, std::vector<LexiconFreeDecoderState>());
+
+  /* note: the lm reset itself with :start() */
+  hyp_[0].emplace_back(0.0, lm_->start(0), nullptr, sil_);
+  nDecodedFrames_ = 0;
+  nPrunedFrames_ = 0;
+}
+
+void LexiconFreeDecoder::decodeStep(const float* emissions, int T, int N) {
+  int startFrame = nDecodedFrames_ - nPrunedFrames_;
+  // Extend hyp_ buffer
+  if (hyp_.size() < startFrame + T + 2) {
+    for (int i = hyp_.size(); i < startFrame + T + 2; i++) {
```

Comment:
not required, but using the same increment operator in for loop looks better.  we can stick to one of `i++` and `++i` in this PR.

Reply:
would prefer to leave the source implementation unchanged, as this is ported directly from flashlight repo and there may be performance implications for changing code. I can make the suggestions to the code comments though.

Reply:
I see, if the source code is directly adopted, it makes sense to keep as what it is.